pkg/apis/core/v1/defaults.go

```diff
@@ -211,9 +218,11 @@ func SetDefaults_PodSpec(obj *v1.PodSpec) {
 	if obj.RestartPolicy == "" {
 		obj.RestartPolicy = v1.RestartPolicyAlways
 	}
-	if obj.HostNetwork {
-		defaultHostNetworkPorts(&obj.Containers)
-		defaultHostNetworkPorts(&obj.InitContainers)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.LatePodHostnetHostPort) {
```

Comment:
if we go with an off-by-default `Deprecated` feature gate name, this would switch polarity, e.g. `if utilfeature.DefaultFeatureGate.Enabled(features.DefaultHostPortsInHostNetworkWorkloadObjects)`